src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -202,10 +203,102 @@ public IRecordingDescriptor startRecording(
+                                    String.format(
+                                            "Recording with name \"%s\" already exists",
+                                            recordingName));
+                        } else {
+                            connection.getService().close(previous.get());
+                        }
+                    }
+                    IRecordingDescriptor desc =
+                            connection
+                                    .getService()
+                                    .start(
+                                            recordingOptions,
+                                            enableEvents(
+                                                    connection,
+                                                    templateName,
+                                                    preferredTemplateType));
+                    String targetId = connectionDescriptor.getTargetId();
+
+                    Map<String, String> labels = metadata.getLabels();
+                    labels.put("template.name", templateName);
+                    labels.put("template.type", preferredTemplateType.name());
+                    Metadata updatedMetadata = new Metadata(labels);
+                    updatedMetadata =
+                            recordingMetadataManager
+                                    .setRecordingMetadata(
+                                            connectionDescriptor, recordingName, updatedMetadata)
+                                    .get();
+                    HyperlinkedSerializableRecordingDescriptor linkedDesc =
+                            new HyperlinkedSerializableRecordingDescriptor(
+                                    desc,
+                                    webServer.get().getDownloadURL(connection, desc.getName()),
+                                    webServer.get().getReportURL(connection, desc.getName()),
+                                    updatedMetadata,
+                                    archiveOnStop);
+                    this.issueNotification(targetId, linkedDesc, CREATION_NOTIFICATION_CATEGORY);
+
+                    Object fixedDuration =
+                            recordingOptions.get(RecordingOptionsBuilder.KEY_DURATION);
+                    if (fixedDuration != null) {
+                        Long delay =
+                                Long.valueOf(fixedDuration.toString().replaceAll("[^0-9]", ""));
+                        scheduleRecordingTasks(
+                                recordingName, delay, connectionDescriptor, archiveOnStop);
+                    }
+
                     return desc;
                 });
     }
 
+    private boolean shouldRestartRecording(String replace, String restart) {
+        if (replace != null) {
+            switch (replace) {
+                case "always":
+                    return true;
+                case "stopped":
+                    return restart == null || restart.equals("true");
```

Comment:
We should check if the recording is stopped here. If it is, return true. Else, return false.

The restart and replace should be separated out. And `replace`, I believe, should take precedence.

Reply:
The `restart` and `replace` parameter should be separated. And `replace`, I believe, should take precedence.

Reply:
should i get rid of restart completely? or this approach of using both but replace to take precedence is expected?